src/Ldap.php

```diff
@@ -177,7 +193,11 @@ public function getLastError(&$errorCode = null, array &$errorMessages = null)
         }
 
         ErrorHandler::start(E_WARNING);
-        ldap_get_option($this->resource, LDAP_OPT_ERROR_STRING, $estr2);
+        if (is_resource($this->resource)) {
+            ldap_get_option($this->resource, LDAP_OPT_ERROR_STRING, $estr2);
+        } else {
```

Comment:
Can we init the `$estr1` here with `getLastError: could not call ldap_error ...` and reassign the result from `ldap_error` in case we have a proper resource? That would eliminate the `else`.